### PR TITLE
Add field

### PR DESCRIPTION
**Pull request by JindrichSusen**

### CODE REVIEW

**Comment by NdubuisiJr** [comment]
Hi @JindrichSusen 
Thanks for taking out the time to contribute to this project.

I have a few changes I would like you to implement. After which, we can merge...

File: TypeExtender/TypeExtender/TypeExtender.cs [hunk 1]
```diff
@@ -194,6 +235,57 @@ public class TypeExtender {
             }
         }
 
+        private void addField(string fieldName, Type type, Type attributeType, object[] attributeValues)
+        {
```
Comment (NdubuisiJr) [inline TypeExtender/TypeExtender/TypeExtender.cs]: For code style consistency, please implement all your code blocks with the first brace ('{') on the same line as the definition.
e.g 
```
private void addField(...) {
}
```

File: TypeExtender/TypeExtender/TypeExtender.cs [hunk 2]
```diff
@@ -173,6 +173,47 @@ public class TypeExtender {
                 addProperty(properties.ElementAt(index).Replace(" ","_"), types.ElementAt(index), null, null, allReadOnly);
             }
         }
+
```
Comment (NdubuisiJr) [inline TypeExtender/TypeExtender/TypeExtender.cs:176]: Most of the public methods have a Generic overload... 

Please implement a generic overload for the AddField function.

File: TypeExtender/TypeExtender.Test/TypeExtenderShould.cs [hunk 3]
```diff
@@ -150,5 +151,61 @@ public class TypeExtenderShould {
             Assert.NotNull(attribute);
             Assert.AreEqual("Jon Snow", attribute.Name);
         }
+
```
Comment (NdubuisiJr) [inline TypeExtender/TypeExtender.Test/TypeExtenderShould.cs:154]: Please add a test for the generic overload requested below...

**Review by NdubuisiJr — CHANGES_REQUESTED** [review]
Please implement the requested changes and we are good to merge

Thanks a lot

**Comment by JindrichSusen** [comment]
Hi. I hope you like the changes :) Thanks

**Comment by NdubuisiJr** [comment]
Looks great...

@JindrichSusen consider starring the project so that others can find it easily.

Thanks a lot.

**Pull request merged**